Improve `toc-depth` in html output

dragonstyle opened the pull request:
The `toc-depth` option will now control how deep html table of contents will display (previously, we always displayed to TOC to a depth of 3). This will control what depth of items are show when the `toc-location` is body. It will control what depth of items will be opened / tracked when scrolling when `toc-location` is left or right.

dragonstyle commented on src/format/html/format-html-bootstrap.ts:
```diff
@@ -275,11 +279,17 @@ function bootstrapHtmlPostprocessor(
-      for (let i = 0; i < nestedUls.length; i++) {
-        const ul = nestedUls[i] as Element;
-        ul.classList.add("collapse");
+      const tocDepth = format.pandoc[kTocDepth] || 3;
```
Is there a better place for me to be defaulting this?

jjallaire replied:
That seems fine to me

jjallaire commented on changelog.md:
```diff
@@ -1,3 +1,5 @@
 
 ## Unreleased
+
+- respect `toc-depth` in the HTML format (bootstrap) rather than always acting as if depth is 3.
```
Changelog should be a separate commit (in case we want to cherry pick back onto v1.0)

dragonstyle committed:
Improve `toc-depth` in html output

The `toc-depth` option will now control how deep html table of contents will display (previously, we always displayed to TOC to a depth of 3). This will control what depth of items are show when the `toc-location` is body. It will control what depth of items will be opened / tracked when scrolling when `toc-location` is left or right.

dragonstyle committed:
Add changelog

jjallaire commented on changelog.md:
```diff
@@ -1,3 +1,5 @@
 # Changelog
 
 ## Unreleased
```
Let's remove the changelog entry here
